File: 02-backend/spring-boot-ecommerce/src/main/java/com/danmil/ecommerce/dto/Purchase.java
package com.danmil.ecommerce.dto;

import com.danmil.ecommerce.entity.*;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Order order;
    private Address shippingAddress;
    private Address billingAddress;
    private Set<OrderItem> orderItems;

}

File: 03-frontend/angular-ecommerce/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";
// Purchase class to represent a purchase in the e-commerce application
export class Purchase {
    order?: Order;
    customer?: Customer;
    shippingAddress?: Address;
    billingAddress?: Address;
    orderItems?: OrderItem[];
}
